Lib/profiling/sampling/sample.py

```diff
@@ -631,6 +632,9 @@ def sample(
         case "flamegraph":
             collector = FlamegraphCollector(skip_idle=skip_idle)
             filename = filename or f"flamegraph.{pid}.html"
+        case "gecko":
```

Comment:
ℹ️ **Logic**

Consider adding a validation step for the `gecko` format similar to other formats. This ensures that any specific requirements or constraints for the `gecko` format are checked before proceeding.